src/chainstate/stacks/db/blocks.rs

```diff
@@ -914,16 +914,10 @@ impl StacksChainState {
 
         // gather
         let mut blobs = vec![];
-        while let Some(row_res) = rows.next() {
-            match row_res {
-                Ok(row) => {
-                    let next_blob: Vec<u8> = row.get(0);
-                    blobs.push(next_blob);
-                }
-                Err(e) => {
-                    return Err(Error::DBError(db_error::SqliteError(e)));
-                }
-            };
+
+        while let Some(row) = rows.next().map_err(|e| db_error::SqliteError(e))? {
```

Comment:
Error handling in a rust destructuring loop 🥇

Reply:
yeah chaining the `map_err` is not ideal. I've been trying a few loop patterns, this one was the most readable one I could come up with - open to something else.

Reply:
I think for anyone familiar with the destructuring and `?` syntax it is readable enough